include/s3select_functions.h

```diff
@@ -180,6 +181,37 @@ struct _fn_sum : public base_function
     virtual void get_aggregate_result(variable *result) { *result = sum ;} 
 };
 
+struct _fn_avg : public base_function
+{
+
+    value sum;
+    float count = 0;
+
+    _fn_avg() : sum(0) { aggregate = true; }
+
+    bool operator()(std::vector<base_statement *> *args, variable *result)
+    {
+        std::vector<base_statement *>::iterator iter = args->begin();
+        base_statement *x = *iter;
+
+        try
+        {
+            sum = sum + x->eval();
+            count = count + 1;
+        }
+        catch (base_s3select_exception &e)
+        {
+            std::cout << "illegal value for aggregation(sum). skipping." << std::endl;
+            if (e.severity() == base_s3select_exception::s3select_exp_en_t::FATAL)
+                throw;
```

Comment:
use /base_s3select_exception/ to throw exception.
its also better to do the same fix for other functions (like sum)

Reply:
makes sense